app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -131,6 +131,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         View view = inflater.inflate(R.layout.fragment_message_list, container,
                 false);
 
+        messageIndex = new SparseArray<>();
```

Comment:
@vishwesh3 nice work :), just a suggestion, for such tricky issues it is often a good practice to include the explaination in code also via comments.